### PR TITLE
[IMP] top_bar: added placeholder to topbar composer

### PR DESCRIPTION
**Pull request by dhrp-odoo**
## Description:

This PR implements the feature of adding a placeholder to the top bar composer.  It disappears automatically when the user clicks inside the composer.

Odoo task ID : [3175327](https://www.odoo.com/web#id=3175327&menu_id=4720&cids=2&action=333&active_id=2328&model=project.task&view_type=form)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/components/composer/composer/composer.ts [hunk 1]
```diff
@@ -86,6 +86,17 @@ css/* scss */ `
       }
     }
 
+    .o-composer[contenteditable="true"]:empty::before {
```
Comment (hokolomopo) [inline src/components/composer/composer/composer.ts:89]:
```suggestion
    .o-composer[contenteditable="true"]:empty:not(:focus)::before {
```

that way you don't need the next css rule with `display: none;`

**Review by hokolomopo — COMMENTED** [review]
Looks good :+1:

File: src/components/composer/composer/composer.ts (hunk 1 above)
Comment (dhrp-odoo) [reply src/components/composer/composer/composer.ts]: Ok, thanks for the suggestion! Changes made

File: src/components/composer/top_bar_composer/top_bar_composer.xml [hunk 2]
```diff
@@ -5,6 +5,7 @@
         focus="props.focus"
         inputStyle="composerStyle"
         onComposerContentFocused="props.onComposerContentFocused"
+        placeholder="placeholderText"
```
Comment (LucasLefevre) [inline src/components/composer/top_bar_composer/top_bar_composer.xml:8]: you can just hardcode the string here
```suggestion
        placeholder="'fx'"
```

File: src/components/composer/composer/composer.ts [hunk 3]
```diff
@@ -86,6 +86,13 @@ css/* scss */ `
       }
     }
 
+    .o-composer[contenteditable="true"]:empty:not(:focus)::before {
```
Comment (LucasLefevre) [inline src/components/composer/composer/composer.ts:89]: I think [contenteditable="true"] is useless
```suggestion
    .o-composer:empty:not(:focus)::before {
```

**Review by LucasLefevre — COMMENTED** [review]
Only minor comments ;)
Nicely done
Thanks

File: src/components/composer/top_bar_composer/top_bar_composer.xml (hunk 2 above)
Comment (dhrp-odoo) [reply src/components/composer/top_bar_composer/top_bar_composer.xml]: Thank you for your input. I have made the changes.

**Comment by dhrp-odoo** [comment]
This task has been merged in [PR#2223](https://github.com/odoo/o-spreadsheet/pull/2223)

**Pull request closed without merging**